feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/repository/GetPhoneNumberUseCase.kt

```diff
@@ -0,0 +1,34 @@
+package com.hedvig.android.odyssey.repository
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.cache.normalized.FetchPolicy
+import com.apollographql.apollo3.cache.normalized.fetchPolicy
+import com.hedvig.android.apollo.OperationResult
+import com.hedvig.android.apollo.graphql.ProfileQuery
+import com.hedvig.android.apollo.graphql.UpdatePhoneNumberMutation
+import com.hedvig.android.apollo.safeExecute
+
+class GetPhoneNumberUseCase(
+  private val apolloClient: ApolloClient,
+) {
+  private val profileQuery = ProfileQuery()
+
+  suspend fun getPhoneNumber(): PhoneNumberResult = apolloClient
+    .query(profileQuery)
+    .fetchPolicy(FetchPolicy.NetworkOnly)
+    .safeExecute()
+    .let {
+      when (it) {
+        is OperationResult.Error -> PhoneNumberResult.Error(it.message ?: "Unknown error")
+        is OperationResult.Success -> PhoneNumberResult.Success(it.data.member.phoneNumber ?: "")
+      }
+    }
+
+  suspend fun updatePhoneNumber(input: String) =
+    apolloClient.mutation(UpdatePhoneNumberMutation(input)).safeExecute()
+}
```

Comment:
Since this does both getting it and updating the PhoneNumber maybe can be renamed to PhoneNumberRepository instead.